### PR TITLE
net: lib: download_client: add support for non-default ports

### PR DESCRIPTION
**Pull request by lemrey**
Let the user specify a port to be used when connecting.

### CODE REVIEW

File: include/net/download_client.h [hunk 1]
```diff
@@ -81,10 +81,17 @@ struct download_client_evt {
  * @brief Download client configuration options.
  */
 struct download_client_cfg {
-	/** TLS security tag. If -1, TLS is disabled. */
+	/** IP port.
+	 *  Pass zero to use default, or non-zero to override.
+	 */
+	int port;
```
Comment (rlubos) [inline include/net/download_client.h]: I'd make it `u16_t`, port numbers are no larger.

File: include/net/download_client.h [hunk 2]
```diff
@@ -81,10 +81,17 @@ struct download_client_evt {
...
+	/** TLS security tag.
+	 * Pass -1 to disable TLS.
```
Comment (rlubos) [inline include/net/download_client.h]: nit: alignment

File: include/net/download_client.h [hunk 3]
```diff
@@ -81,10 +81,17 @@ struct download_client_evt {
...
+	 */
 	int sec_tag;
 	/** Access point name identifying a packet data network.
-	 * Must be a null-terminated string or NULL to use the default APN.
+	 * Pass a null-terminated string with the APN
```
Comment (rlubos) [inline include/net/download_client.h]: nit: alignment

File: include/net/download_client.h (hunk 1 above)
Comment (lemrey) [reply include/net/download_client.h]: Right, I used `u16_t` in `resolve_and_connect()` but forgot it here.

File: include/net/download_client.h (hunk 2 above)
Comment (lemrey) [reply include/net/download_client.h]: The `*` alignment looks correct to me (`checkpatch` will complain if I align it with the second asterisk on the line above).

File: include/net/download_client.h (hunk 3 above)
Comment (lemrey) [reply include/net/download_client.h]: Ditto

File: include/net/download_client.h (hunk 2 above)
Comment (lemrey) [reply include/net/download_client.h]: I see now which alignment you meant :)

**Pull request merged**